client/test/components/notebook/convert.test.ts

```diff
@@ -0,0 +1,250 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  generateCodeListFromPythonNotebook,
+  generateCodeListFromSASNotebook,
+  generateFlowDataNode,
+  generateFlowDataSwimlane,
+} from "../../../src/components/ContentNavigator/convert";
+import * as assert from "assert";
+
+const contentSASNotebook1 = `[{"language": "python","value": "a, b = 4, 2\r\nprint('Result: ', a*10 + b)"},{"language": "sas","value": "data work.prdsale;\r\n\tset sashelp.PRDSALE;\r\nrun;\r\n\r\nproc means data=work.prdsale;\r\nrun;"},{"language": "markdown","value": "# test"},{"language": "sql","value": "CREATE TABLE WORK.QUERY_PRDSALE AS\r\n    SELECT\r\n        (t1.COUNTRY) LABEL='Country' FORMAT=$CHAR10.,\r\n        (SUM(t1.ACTUAL)) FORMAT=DOLLAR12.2 LENGTH=8 AS SUM_ACTUAL\r\n    FROM\r\n        WORK.PRDSALE t1\r\n    GROUP BY\r\n        t1.COUNTRY"}]`;
```

Comment:
It may be more readable to start with these being json objects, and stringify as needed for tests. And then we can use backticks for strings and use tab/newlines/etc as needed. So we could do something like...

```javascript
const contentSASNotebook1 = [{
	"language": "python",
	"value": `a, b = 4, 2
print('Result: ', a*10 + b)`
}, {
	"language": "sas",
	"value": `data work.prdsale;
		set sashelp.PRDSALE;
run;

proc means data=work.prdsale;
run;`
}, {
	"language": "markdown",
	"value": "# test"
}, {
	"language": "sql",
	"value": `CREATE TABLE WORK.QUERY_PRDSALE AS
SELECT
(t1.COUNTRY) LABEL='Country' FORMAT=$CHAR10.,
(SUM(t1.ACTUAL)) FORMAT=DOLLAR12.2 LENGTH=8 AS SUM_ACTUAL
FROM
WORK.PRDSALE t1
GROUP BY
t1.COUNTRY`
}];
```

and that would change
```javascript
contentSASNotebook1
        .replace(/\n/g, "\\n")
        .replace(/\t/g, "\\t")
        .replace(/\r/g, "\\r"),
```

to something more like `JSON.stringify(contentSASNotebook1)` (newlines/tabs/etc should be properly encoded without replacements)

Reply:
done in 5f3a455e8798a4a89997b40e894f660e63b00135